dev-client/src/components/sites/CreateSiteView.tsx

```diff
@@ -107,13 +110,21 @@ export default function CreateSiteView({
       }
     }
     const {name, latitude, longitude, projectId} = validationResults;
-    createSiteCallback({name, latitude, longitude, projectId});
-    if (defaultProject) {
-      goBack();
-      return;
+    const createdSite = await createSiteCallback({
+      name,
+      latitude,
+      longitude,
+      projectId,
+    });
+    if (createdSite !== undefined) {
+      reset({
+        routes: [
+          {name: 'HOME'},
+          {name: 'LOCATION_DASHBOARD', params: {siteId: createdSite.id}},
+        ],
+      });
```

Comment:
question: i didn't know the reset function existed! when i was doing a similar change to navigate to the project page after creating a project, i used the `replace` function for similar reasons. do you know what the exact expected behavior is for navigation? e.g. if you create a site from a project's sites view, you'll now be taken to the site's dashboard. but if you exist the dashboard, should you go back to the project's sites view or back to the home screen?

Reply:
Yep, using `replace` seems like a better solution here for sure!